171293032 planta carnivora

JuanmaSalva opened the pull request:
Cosas añadidas:
Clase carnivourousPlant
Animación cutre

Como funciona:
hay un temporizador en milisegundos que al llegar a 0 te come. Se puede ver como va quedando menos tiempo por que la animacion idle de la planta va cada vez mas rápido. Si entras, pasas 1 segundo y sales, cuando vuelvas a entrar el tiempo sigue desde ese segundo, no se resetea. Si hay dos jugadores la hostia se la lleva el ultimo en entrar por imbécil.
Aparece en el GYM por q no tenemos jardín pero va con tiled asi q es solo cambiarlo desde alli.

Creo q eso es todo, tengo pesadillas con esto

JuanmaSalva committed:
Animacion hecha

hay un temporizador y acelera hasta que te mata

JuanmaSalva committed:
te intenta matar pero se queda en el intento

JuanmaSalva committed:
Merge branch 'master' into 171293032-planta-carnivora

JuanmaSalva committed:
Merged master

JuanmaSalva committed:
Merge branch 'master' into 171293032-planta-carnivora

JuanmaSalva committed:
socorro no funciona

JuanmaSalva committed:
Funciona pa 1 jugador

JuanmaSalva committed:
funciona pera la hostia se la lleva el primero q entra jeje

JuanmaSalva committed:
terminado, todo funciona

creo

EstebenR commented on src/config/constants.json:
```diff
@@ -241,6 +241,17 @@
+  "CARNIVOROUSEPLANT_MAX_SPEED": 4,
+  "IDLE_FRAMES": 2,
+  "EATING_FRAMES": 5,
+  "FRAME_SIZE": 32,
```
Estos nombres deberían ir tipo CarnivorousPlant_size etc
btw carnivorous no acaba en e

EstebenR commented on src/SourceFiles/CarnivorousPlant.h:
```diff
@@ -0,0 +1,33 @@
+	Viewer* viewer_;
+	int frame_, minAnimationSpeed_, maxAnimationSpeed_, playersInside_,
+		idleFrames_, eatingFrames_, frameSize_, coinDMG, damage;
+	float aumento_, actualSpeed_, timePassed_, enterTime_, limitTime_;
```
aumento deberia ir en ingles

EstebenR reviewed (changes requested):
Mejor hacer que las medidas de tiempo vayan con frames en vez de getTime o dependa de que cada frame vaya a 16.66

JuanmaSalva committed:
cambio de tiempo a frames & fix varios

las constantes ahora estan bien escritas, se han quitado todas las variables/metodos relacionados con el control de timpo

eljorgemo commented on src/SourceFiles/CarnivorousPlant.cpp:
```diff
@@ -0,0 +1,90 @@
+		else frame_ = 0;
+	}
+	else {
+		int pos = frameSize_ * (frame_ / (int)actualSpeed_) + frameSize_ * idleFrames_;
```
frame/actualSpeed puede ponerse multiplicando? las divisiones cuestan muchos ciclos. O al menos se podría cambiar el sitio donde se calcula la división para que se haga sólo cuando cambia el actualSpeed_ o el frame

eljorgemo reviewed (approved):
Okey. Hay que revisar los tamaños de los colliders

JuanmaSalva committed:
Merge branch 'master' into 171293032-planta-carnivora
